File: wimg-ios/wimg/Models/Summary.swift
import Foundation

struct MonthlySummary: Codable {
    let year: Int
    let month: Int
    let income: Double
    let expenses: Double
    let available: Double
    let tx_count: Int
    let by_category: [CategoryBreakdown]
}

struct CategoryBreakdown: Codable, Identifiable {
    let id: Int
    let name: String
    let amount: Double
    let count: Int
}

File: wimg-android/app/src/main/java/com/wimg/app/models/Summary.kt
package com.wimg.app.models

import kotlinx.serialization.Serializable

@Serializable
data class MonthlySummary(
    val year: Int,
    val month: Int,
    val income: Double,
    val expenses: Double,
    val available: Double,
    val tx_count: Int,
    val by_category: List<CategoryBreakdown> = emptyList(),
)

@Serializable
data class CategoryBreakdown(
    val id: Int,
    val name: String,
    val amount: Double,
    val count: Int,
)
